### PR TITLE
Enabled most unit tests for ESP32

### PR DESCRIPTION
**Pull request by feasel0**
This has been replaced by [PR 36738](https://github.com/project-chip/connectedhomeip/pull/36738) in order to correct the commit message

### CODE REVIEW

**Comment by lboue** [comment]
Are you sure about your PR title?

**Review by andy31415 — APPROVED** [review]

**Comment by shubhamdp** [comment]
due to this check, the tests won't run in the CI. Should we remove that `github.repository_owner == 'espressif'` check from there so that they'll run on every PR? @andy31415 @shripad621git 
https://github.com/project-chip/connectedhomeip/blob/29c264723be6ebfb1564d5f999f96e2f70cbc1e0/.github/workflows/qemu.yaml#L41

File: config/esp32/components/chip/idf_component.yml [hunk 1]
```diff
@@ -41,3 +41,5 @@ dependencies:
         version: ">=0.1.12"
         rules:
             - if: "target in [esp32p4]"
+
+espressif/esp_diag_data_store: "1.0.1"
```
Comment (dhrishi) [inline config/esp32/components/chip/idf_component.yml:45]: Hi @shripad621git Should we pin the version of esp_diag_data_store here or esp_insights?

File: src/platform/ESP32/ConfigurationManagerImpl.h [hunk 2]
```diff
@@ -35,6 +35,7 @@
 #include <platform/internal/GenericConnectivityManagerImpl_NoBLE.h>
 #endif
 
+#include "freertos/timers.h" // TimerHandle_t
```
Comment (dhrishi) [inline src/platform/ESP32/ConfigurationManagerImpl.h:38]: @feasel0 Why is this required here? Any compilation errors you see without this?

File: config/esp32/components/chip/idf_component.yml (hunk 1 above)
Comment (shripad621git) [reply config/esp32/components/chip/idf_component.yml:45]: The above change is not required. We have pinned the insights versions [here](https://github.com/espressif/esp-insights/commit/a417202e7f2dce53853f46a87393a614e17b32e3) and we are using the latest esp-insights i.e 1.2.2.
The flow with insights enabled is tested with 1.2.2.

**Comment by shripad621git** [comment]
> due to this check, the tests will be skipped in the CI. Should we remove that `github.repository_owner == 'espressif'` check from there so that they'll run on every PR? @andy31415 @shripad621git
> 
> https://github.com/project-chip/connectedhomeip/blob/29c264723be6ebfb1564d5f999f96e2f70cbc1e0/.github/workflows/qemu.yaml#L41

If we want to run the qemu tests on the PR itself, the check should  be removed. Also the CI related to QEMU should be tested on this PR before merging this. So , even if we wish to skip the test and run it on espressif/connectedhomeip CI, the changes should specific to QEMU should be tested on this PR.

File: config/esp32/components/chip/idf_component.yml (hunk 1 above)
Comment (dhrishi) [reply config/esp32/components/chip/idf_component.yml:45]: @feasel0 PTAL

File: src/platform/ESP32/ConfigurationManagerImpl.h (hunk 2 above)
Comment (feasel0) [reply src/platform/ESP32/ConfigurationManagerImpl.h]: The error (undefined reference to `TimerHandle_t`) seems to have gone away after the recent version update.  I've removed this line, since it's no longer needed.

**Comment by feasel0** [comment]
I removed the condition, enabling the two esp32 workflows, and confirmed that this PR does pass CI.  Before merging this, let me know if we want to have it this way (runs on all PRs) or revert it to the previous state (only runs on PRs from esspressif).
